### PR TITLE
perf(markdown): do not run change detection when the markdown is clicked

### PR DESCRIPTION
**Pull request by arturovt**
#### General Tests for Every PR

- [x] `npm run serve:prod` still works.
- [x] `npm run tslint` passes.
- [x] `npm run stylelint` passes.
- [x] `npm test` passes and code coverage is not lower.
- [x] `npm run build:lib` still works.

### CODE REVIEW

**Comment by owilliams320** [comment]
@arturovt looks like the tests are not passing

File: src/platform/markdown/markdown.component.ts [hunk 1]
```diff
@@ -224,6 +223,27 @@ export class TdMarkdownComponent implements OnChanges, AfterViewInit {
...
+          if (element.matches('a[href]') && isAnchorLink(<HTMLAnchorElement>element)) {
+            this.handleAnchorClicks(event);
+          }
+        },
+      );
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._removeClickListenerFn?.();
```
Comment (owilliams320) [inline src/platform/markdown/markdown.component.ts:246]: can we just name this something like `anchorListener`

**Review by owilliams320 — CHANGES_REQUESTED** [review]

**Comment by arturovt** [comment]
@owilliams320 had to run run prettier locally since the lint job was failing.

**Review by owilliams320 — APPROVED** [review]
LGTM 🚀

**Pull request merged**